crates/core/src/lib.rs

```diff
@@ -34,31 +34,48 @@ pub extern "C" fn init() {
 
         let _ = context.eval_global(SCRIPT_NAME, &contents).unwrap();
         let global = context.global_object().unwrap();
-        let shopify = global.get_property("Shopify").unwrap();
-        let main = shopify.get_property("main").unwrap();
+        let suborbital = global.get_property("Suborbital").unwrap();
+        let main = suborbital.get_property("main").unwrap();
 
         JS_CONTEXT.set(context).unwrap();
-        ENTRYPOINT.0.set(shopify).unwrap();
+        ENTRYPOINT.0.set(suborbital).unwrap();
         ENTRYPOINT.1.set(main).unwrap();
     }
 }
 
-#[export_name = "shopify_main"]
-pub extern "C" fn run() {
-    unsafe {
-        let context = JS_CONTEXT.get().unwrap();
-        let shopify = ENTRYPOINT.0.get().unwrap();
-        let main = ENTRYPOINT.1.get().unwrap();
-        let input_bytes = engine::load().expect("Couldn't load input");
-
-        let input_value = transcode_input(&context, &input_bytes).unwrap();
-        let output_value = main.call(&shopify, &[input_value]);
+struct JsRunnable {}
 
-        if output_value.is_err() {
-            panic!("{}", output_value.unwrap_err().to_string());
+impl Runnable for JsRunnable {
+    fn run(&self, in_bytes: Vec<u8>) -> Result<Vec<u8>, RunErr> {
+        unsafe {
+            let context = JS_CONTEXT.get().unwrap();
+            let suborbital = ENTRYPOINT.0.get().unwrap();
+            let main = ENTRYPOINT.1.get().unwrap();
+            let input = context
+                .value_from_bytes(in_bytes)
+                .expect("Couldn't load input");
+            let output_value = main.call(&suborbital, &[input]);
```

Comment:
This is more or less the same API as JS' `Function.prototype.call`, where the first argument is the `this` value. In our case, it can probably just be null.

Reply:
Gotcha. Rust likely won't let you pass null here, so it's fine as-is.